### ISSUE TITLE
PHP Warning:  Undefined array keys at 1.6.0

### ISSUE THREAD

**Issue by 3kami3** [ltb-project/self-service-password#909]
I upgraded to 1.6.0 and a PHP Warning was logged.
Do I need to configure anything additional in config.inc.local.php?

```
[Thu May 23 21:07:31.042136 2024] [php:warn] [pid 22] [client 172.18.0.2:56836] PHP Warning:  Undefined array key "extended_error_msg" in /var/www/templates_c/286e41dd2c3e04c2d0efd05cb44c5af70eb8f2aa_2.file.index.tpl.php on line 67, referer: https://my.domain.com/index.php
[Thu May 23 21:07:31.042168 2024] [php:warn] [pid 22] [client 172.18.0.2:56836] PHP Warning:  Attempt to read property "value" on null in /var/www/templates_c/286e41dd2c3e04c2d0efd05cb44c5af70eb8f2aa_2.file.index.tpl.php on line 67, referer: https://my.domain.com/index.php
[Thu May 23 21:07:31.046295 2024] [php:warn] [pid 22] [client 172.18.0.2:56836] PHP Warning:  Undefined array key "attribute_mail_update" in /var/www/templates_c/181bd598c64907a17489511d59d14633c2561c5b_2.file.sendtoken.tpl.php on line 35, referer: https://my.domain.com/index.php
[Thu May 23 21:07:31.046313 2024] [php:warn] [pid 22] [client 172.18.0.2:56836] PHP Warning:  Attempt to read property "value" on null in /var/www/templates_c/181bd598c64907a17489511d59d14633c2561c5b_2.file.sendtoken.tpl.php on line 35, referer: https://my.domain.com/index.php
[Thu May 23 21:07:42.092700 2024] [php:warn] [pid 21] [client 172.18.0.2:51112] PHP Warning:  Undefined array key "msg_changehelpextramessage" in /var/www/templates_c/3ba7131c51edfe0cef48341f1409d3c1a60d3c16_2.file.change.tpl.php on line 43, referer: https://my.domain.com/index.php?action=sendtoken
[Thu May 23 21:07:42.092724 2024] [php:warn] [pid 21] [client 172.18.0.2:51112] PHP Warning:  Attempt to read property "value" on null in /var/www/templates_c/3ba7131c51edfe0cef48341f1409d3c1a60d3c16_2.file.change.tpl.php on line 43, referer: https://my.domain.com/index.php?action=sendtoken
```

* Version: 1.6.0
* Platform: Official Docker image

**Comment by davidcoutadeur** [comment]
Hello @3kami3 

You can:
1. verify that the config.inc.php has been updated with the last version of self-service-password
2. empty your cache /var/www/templates_c/

**Comment by JoeDat** [comment]
I have the same issue.  However, clearing templates_c is causing the site to load to a blank page.  I'd suggest moving or renaming the files first just in case you run into something similar.  I've since restored mine from backup.  Seeing the following error after clearing.  Not sure if my custom path to SSP is to blame.  

```
[23-May-2024 14:56:36 America/New_York] PHP Warning:  require_once(/opt/self-service-password/conf/config.inc.php): Failed to open stream: Permission denied in /opt/self-service-password/htdocs/index.php on line 11
[23-May-2024 14:56:36 America/New_York] PHP Fatal error:  Uncaught Error: Failed opening required '../conf/config.inc.php' (include_path='.:/usr/share/pear:/usr/share/php') in /opt/self-service-password/htdocs/index.php:11
Stack trace:
#0 {main}
  thrown in /opt/self-service-password/htdocs/index.php on line 11
```

**Comment by radokristof** [comment]
I observe the same error on version 1.6
Downgrading to v1.5 solves the issue.

**Comment by coudot** [comment]
See #912, same issue?

**Comment by radokristof** [comment]
I don't think it is the same issue. In my case errors are not visible on UI and not related to password changing.

**Comment by davidcoutadeur** [comment]
I reproduce the warnings, at least for the last two messages:

```
[Wed Jun 19 08:13:00.291547 2024] [php:warn] [pid 19] [client 172.17.0.1:58980] PHP Warning:  Undefined array key "msg_changehelpextramessage" in /var/www/templates_c/3ba7131c51edfe0cef48341f1409d3c1a60d3c16_2.file.change.tpl.php on line 43
[Wed Jun 19 08:13:00.291562 2024] [php:warn] [pid 19] [client 172.17.0.1:58980] PHP Warning:  Attempt to read property "value" on null in /var/www/templates_c/3ba7131c51edfe0cef48341f1409d3c1a60d3c16_2.file.change.tpl.php on line 43
```

These messages are visible:
- in any distribution : tarball, docker image, or debian package
- the only condition for displaying them is to have `$debug = true` in your configuration file. In this case, smarty complains that some variables are not initialized, which is no a big deal IMO
- depending on your smarty version, you can also have many more messages in debug mode

I'll search for a solution for not displaying these messages, but it does not seem very important.

**Comment by davidcoutadeur** [comment]
Fix introduced in f240685 has also produced many more messages in smarty (including useless notice)

I have made a pull request to remove smarty messages (except errors) unless `$smarty_debug == true`. See 3d09783

This should also remove all your warnings here.
